maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -737,34 +771,62 @@ protected void configure()
         toolchainsBuilder = container.lookup( ToolchainsBuilder.class );
 
         dispatcher = (DefaultSecDispatcher) container.lookup( SecDispatcher.class, "maven" );
+    }
 
-        return container;
+    protected AbstractModule createModule( CoreExports exports )
+    {
+        return new AbstractModule()
+        {
+            @Override
+            protected void configure()
+            {
+                bind( ILoggerFactory.class ).toInstance( slf4jLoggerFactory );
+                bind( CoreExports.class ).toInstance( exports );
+            }
+        };
     }
 
-    private List<CoreExtensionEntry> loadCoreExtensions( CliRequest cliRequest, ClassRealm containerRealm,
-                                                         Set<String> providedArtifacts )
+    protected List<CoreExtension> loadCoreExtensionsDescriptors( File multiModuleProjectDirectory )
```

Comment:
This name confuses me. Internally `readCoreExtensionsDescriptor` is called, yet this method ends with a plural. Why?

Reply:
Yes, I am not sure what's the best way here.  I went to the minimal changes, but I agree it's far from a good naming.
I can rename `loadCoreExtensions` to `loadCoreExtensionsEntries` and `loadCoreExtensionsDescriptors` to `loadCoreExtensions`.

Reply:
The methods have been renamed to match the return type better.